index.ts

```diff
@@ -42,39 +28,55 @@ async function indexSignaturesForAddress(
   until?: TransactionSignature
 ) {
   let sigs: ConfirmedSignatureInfo[];
+  let latest = before;
+  let earliest = until;
   do {
     sigs = await rpc.getSignaturesForAddressWithRetries(address, {
-      before,
-      until,
+      before: latest,
+      until: earliest,
       limit: 1000,
     });
     sigs.reverse();
+    // Edge case where you enter indexing but there's nothing to index
+    if (!sigs || sigs.length === 0) {
+      // Edge case when before is undefined and there are no new txes to pull from rpc
+      return { earliest: "", latest: before ? before : until };
+    }
+
+    latest = sigs[sigs.length - 1].signature;
+    earliest = sigs[0].signature;
+    // on the first fetch set the latest tx sig (because from here we go back in time)
+    if (before === undefined) {
+      before = latest;
+    }
+
+    if (sigs[0] == undefined || !sigs[sigs.length - 1] == undefined) {
```

Comment:
Never null sigs in between I guess this assumes?

Reply:
Yeah honestly this code doesn't really do anything and was more of a quick hacky check for null sigs since I log the tx range in the below lines. Can remove or do a map checking for null signatures if we want.